xbmc/video/TeletextDefines.h

```diff
@@ -422,6 +424,8 @@ typedef struct TextCacheStruct_t
   unsigned short *ColorTable;
 
   std::string      line30;
+
+  CCriticalSection m_critSection;
```

Comment:
I am not a big fan of this. Exposing a mutex publicly in some random pod structure is a sign of a bad smell.
Personally I feel the solution here is to create a proper cache class (instead of this struct) redefining the several operators, internally protected by its own private mutex.
Unfortunately this is still a bit of work. Maybe we can have this for now adding a `//! TODO` above this line stating we should get rid of the public mutex.

Reply:
I agree with you.